Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2158,14 +2200,11 @@ void WebProcessProxy::didExceedCPULimit()
         }
     }
 
-#if PLATFORM(MAC) && USE(RUNNINGBOARD)
     // This background WebProcess is using too much CPU so we try to suspend it if possible.
-    if (runningBoardThrottlingEnabled() && !throttler().isSuspended() && !isRunningServiceWorkers()) {
```

Comment:
It does not seem like `tryToForceSuspend()` is performing these checks. Is it ok to try to suspend unconditionally?

Reply:
It should be okay because if the process is running service workers, then it should be holding on to some assertion already which will cause `tryToForceSuspend()` to return false.